java/statestore/src/main/java/sleeper/statestore/dynamodb/DynamoDBFileReferenceStore.java

```diff
@@ -178,6 +194,36 @@ public void splitFileReferences(List<SplitFileReferenceRequest> splitRequests) t
                 LOGGER.debug("Applying {} split reference requests", batch.getRequests().size());
                 applySplitRequestWrites(batch, updateTime);
             }
+        } catch (TransactionCanceledException e) {
+            List<CancellationReason> cancellationReasons = e.getCancellationReasons();
+            int transactionIndex = 0;
+            StateStoreException cause = null;
+            for (SplitFileReferenceRequest request : batch.getRequests()) {
+                CancellationReason deleteReason = cancellationReasons.get(transactionIndex);
+                if (isConditionCheckFailure(deleteReason)) {
+                    if (deleteReason.getItem() != null) {
+                        FileReference failedUpdate = fileReferenceFormat.getFileReferenceFromAttributeValues(deleteReason.getItem());
+                        if (failedUpdate.getJobId() != null) {
+                            cause = new FileReferenceAssignedToJobException(failedUpdate);
+                        } else {
+                            cause = new FileReferenceNotFoundException(failedUpdate);
+                        }
+                    } else {
+                        cause = new FileReferenceNotFoundException(request.getOldReference());
+                    }
+                }
+                List<CancellationReason> writeReasons = cancellationReasons.subList(transactionIndex + 1,
+                        transactionIndex + 1 + request.getNewReferences().size());
+                for (int writeReasonIndex = 0; writeReasonIndex < writeReasons.size(); writeReasonIndex++) {
+                    if (isConditionCheckFailure(writeReasons.get(writeReasonIndex))) {
+                        cause = new FileReferenceAlreadyExistsException(request.getNewReferences().get(writeReasonIndex));
+                    }
+                }
+                transactionIndex += 1 + request.getNewReferences().size();
+            }
```

Comment:
This can be its own method to process the cancellation reasons, or split further.